### ISSUE TITLE
tag support in vpc_endpoint datasource

### ISSUE THREAD

**Issue by kimor79** [terraform-providers/terraform-provider-aws#10465]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The current vpc_endpoint datasource only accepts an id, service name, state, or VPC id. If there are multiple endpoints for the same service (ie with different policies), this resource can't be used to select the desired one. Nor can it work properly when there are multiple endpoints for the same service: `multiple VPC Endpoints matched; use additional constraints to reduce matches to a single VPC Endpoint`

It would be nice if we could pass tags to filter on to identify which endpoint we want.

### New or Affected Resource(s)

* aws_vpc_endpoint datasource

### Potential Terraform Configuration

```hcl
data "aws_vpc_endpoint" "bar" {
  service_name = "com.amazonaws.${data.aws_region.current.name}.execute-api"
  tags {
    Name = "bar"
  }
}

data "aws_vpc_endpoint" "foo" {
  service_name = "com.amazonaws.${data.aws_region.current.name}.execute-api"
  tags {
    Name = "foo"
  }
}
```

**Comment by ewbankkit** [comment]
When the [initial support](https://github.com/terraform-providers/terraform-provider-aws/issues/8625) for VPC Endpoint tagging was implemented I don't think the [`DescribeVpcEndpoints`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcEndpoints.html) API method supported tag filters.
It looks like tag filters are now supported.
I can implement.

**Comment by bflad** [comment]
Support for filtering by `filter` and `tags` arguments has been merged and will release with version 2.35.0 of the Terraform AWS Provider, Thursday next week. Thanks to @ewbankkit for the implementation. 👍
